test/framework/src/main/java/org/elasticsearch/test/rest/RestActionTestCase.java

```diff
@@ -68,6 +69,8 @@ protected void dispatchRequest(RestRequest request) {
         ThreadContext threadContext = verifyingClient.threadPool().getThreadContext();
         try (ThreadContext.StoredContext ignore = threadContext.stashContext()) {
             controller.dispatchRequest(request, channel, threadContext);
+        } finally {
+            Releasables.close(channel.capturedResponse());
```

Comment:
This is my only concern in this PR. Its weird to make such a broadly impacting change just for some other test suites. But, from what I can tell, this should have always been this way once: https://github.com/elastic/elasticsearch/pull/65921 was merged in the long long ago. 

Am I correct in thinking this? Or is this change solely because of the async search test failures?